### PR TITLE
Feat/accessibility v2

### PR DESCRIPTION
**Pull request by ivonne-hernandez**
What (if any) features are you implementing?
- Resolve all of the accessibility issues with the exception of the landmark issue for the citizenship filters and "Save My Results" components.
- I added a close button to the filters so that all users are able to close out of the component.

Were there any issues that arose?
- Given that our Modal components are positioned absolutely, I was not able to resolve the `All page content should be contained by landmarks` issue for them.

Is there anything that you need from your teammate?
- Please review and let me know if you have any questions or concerns.

Updated citizenship filters with a close button:
Screenshot 2024-05-14 at 11 47 57 AM
Screenshot 2024-05-14 at 11 50 01 AM

Save my results accessibility issues before and after:
Screenshot 2024-05-14 at 10 31 16 AM
Screenshot 2024-05-14 at 10 31 33 AM

Filter accessibility issues before and after:
Screenshot 2024-05-14 at 10 30 18 AM
Screenshot 2024-05-14 at 10 32 01 AM

### CODE REVIEW

**Comment by CalebPena** [comment]
Does the share modal also have accessibility issues?

**Comment by ivonne-hernandez** [comment]
> Does the share modal also have accessibility issues?

I didn't review the share modal since we're pulling the share component out of the modal altogether and putting it in the footer.

**Review by CalebPena — CHANGES_REQUESTED** [review]

File: src/App.css [hunk 1]
```diff
@@ -88,8 +88,8 @@
   font-size: 1rem;
 }
 
-.sign-up-data-privacy-link {
-  color: var(--primary-color);
+.link-color {
+  color: var(--link-color);
```
[the remark this answers is not in the document]
Comment (ivonne-hernandez) [reply src/App.css:92]: Yes. The contrast ratio of the primary color against the pink/lavender background wasn't high enough so we swapped it out for a different color.

File: src/Components/HouseholdDataBlock/HouseholdDataBlock.js [hunk 2]
```diff
@@ -232,8 +237,8 @@ const HouseholdDataBlock = ({ handleHouseholdDataSubmit }) => {
         <div className="household-member-header">
           <h3 className="member-added-relationship">{relationship}:</h3>
           <div className="household-member-edit-button">
-            <IconButton onClick={handleEditSubmit}>
-              <EditIcon />
+            <IconButton onClick={handleEditSubmit} aria-label={editHHMemberAriaLabel}>
+              <EditIcon alt="edit icon" />
```
[the remark this answers is not in the document]
Comment (ivonne-hernandez) [reply src/Components/HouseholdDataBlock/HouseholdDataBlock.js:241]: Yes I can on a different ticket.

File: src/Components/Results/Programs/Filter.tsx [hunk 3]
```diff
@@ -132,29 +134,41 @@ export const Filter = () => {
...
-            setCitButtonClass(updatedCitButtonClass);
-          }}
+          onClose={handleFilterClose}
           anchorEl={citizenshipPopoverAnchor}
           anchorOrigin={{
             vertical: 'bottom',
             horizontal: 'left',
           }}
           transformOrigin={{ vertical: 2, horizontal: 0 }}
+          aria-labelledby="citizenship-filters-modal"
```
[the remark this answers is not in the document]
Comment (ivonne-hernandez) [reply src/Components/Results/Programs/Filter.tsx:165]: It links the modals to the components that it renders.

File: src/Assets/styleController.ts [hunk 4]
```diff
@@ -55,6 +56,7 @@ const themes: IThemes = {
       '--icon-color': '#D6743F',
       '--secondary-icon-color': '#D6743F',
       '--option-card-hover-font-color': '#1D1C1E',
+      '--link-color': '#41528C',
```
[the remark this answers is not in the document]
Comment (ivonne-hernandez) [reply src/Assets/styleController.ts:59]: Good call. I'll go and change them.

File: src/Components/HouseholdDataBlock/HouseholdDataBlock.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ivonne-hernandez) [reply src/Components/HouseholdDataBlock/HouseholdDataBlock.js:241]: @sydney-devine Can you please create a ticket to have all of the `alt` text translated?

File: src/Components/Results/ProgramPage/ProgramPage.tsx [hunk 5]
```diff
@@ -143,7 +143,7 @@ const ProgramPage = ({ program }: ProgramPageProps) => {
                         <a
                           href={`tel:${navigator.phone_number}`}
                           className="phone-link"
-                          style={{ color: 'var(--link-color)' }}
+                          style={{ color: 'var(--midBlue-color)' }}
```
Comment (CalebPena) [inline src/Components/Results/ProgramPage/ProgramPage.tsx:146]:
```suggestion
                class="link-color"
```

**Review by CalebPena — APPROVED** [review]

**Pull request merged**